examples/simple_intersection/example_intersection.cpp

```diff
@@ -15,153 +15,61 @@
-      for (int j = 0; j < ny; ++j)
-        for (int k = 0; k < nz; ++k)
-        {
-          ArborX::Point p_lower{
-              {(i - .25) * hx, (j - .25) * hy, (k - .25) * hz}};
-          ArborX::Point p_upper{
-              {(i + .25) * hx, (j + .25) * hy, (k + .25) * hz}};
-          boxes_host[index(i, j, k)] = {p_lower, p_upper};
-        }
-    Kokkos::deep_copy(execution_space, _boxes, boxes_host);
-  }
-
-  // Return the number of boxes.
-  KOKKOS_FUNCTION int size() const { return _boxes.size(); }
-
-  // Return the box with index i.
-  KOKKOS_FUNCTION ArborX::Box const &get_box(int i) const { return _boxes(i); }
-
-private:
-  Kokkos::View<ArborX::Box *, typename DeviceType::memory_space> _boxes;
-};
-
-// For creating the bounding volume hierarchy given a Boxes object, we
-// need to define the memory space, how to get the total number of objects,
-// and how to access a specific box. Since there are corresponding functions in
-// the Boxes class, we just resort to them.
-template <typename DeviceType>
-struct ArborX::AccessTraits<Boxes<DeviceType>, ArborX::PrimitivesTag>
-{
-  using memory_space = typename DeviceType::memory_space;
-  static KOKKOS_FUNCTION int size(Boxes<DeviceType> const &boxes)
-  {
-    return boxes.size();
-  }
-  static KOKKOS_FUNCTION auto get(Boxes<DeviceType> const &boxes, int i)
-  {
-    return boxes.get_box(i);
-  }
-};
-
-// For performing the queries given a Boxes object, we need to define memory
-// space, how to get the total number of queries, and what the query with index
-// i should look like. Since we are using self-intersection (which boxes
-// intersect with the given one), the functions here very much look like the
-// ones in ArborX::AccessTraits<Boxes<DeviceType>, ArborX::PrimitivesTag>.
-template <typename DeviceType>
-struct ArborX::AccessTraits<Boxes<DeviceType>, ArborX::PredicatesTag>
-{
-  using memory_space = typename DeviceType::memory_space;
-  static KOKKOS_FUNCTION int size(Boxes<DeviceType> const &boxes)
-  {
-    return boxes.size();
-  }
-  static KOKKOS_FUNCTION auto get(Boxes<DeviceType> const &boxes, int i)
-  {
-    return intersects(boxes.get_box(i));
-  }
-};
-
-// Now that we have encapsulated the objects and queries to be used within the
-// Boxes class, we can continue with performing the actual search.
 int main(int argc, char *argv[])
 {
   Kokkos::ScopeGuard guard(argc, argv);
 
   using ExecutionSpace = Kokkos::DefaultExecutionSpace;
-  using MemorySpace = typename ExecutionSpace::memory_space;
-  using DeviceType = Kokkos::Device<ExecutionSpace, MemorySpace>;
-  ExecutionSpace execution_space;
+  using MemorySpace = ExecutionSpace::memory_space;
+
+  Kokkos::View<ArborX::Box *, MemorySpace> boxes("Example::boxes", 4);
+  auto boxes_host = Kokkos::create_mirror_view(boxes);
+  boxes_host[0] = {{0, 0, 0}, {1, 1, 1}};
+  boxes_host[1] = {{1, 0, 0}, {2, 1, 1}};
+  boxes_host[2] = {{0, 1, 0}, {1, 2, 1}};
+  boxes_host[3] = {{1, 1, 0}, {2, 2, 1}};
```

Comment:
Did you consider having an ASCII drawing for the boxes ordering?

Reply:
I did, and decided not to, as for the given point distribution it is obvious which box numbers they would belong to.